great_expectations/core/expectation_suite.py

```diff
@@ -144,31 +144,40 @@ def name(self) -> str:
 
     @property
     def expectations(self) -> list[Expectation]:
-        return [
+        expectations = [
             self._build_expectation(expectation_configuration=expectation_configuration)
             for expectation_configuration in self.expectation_configurations
         ]
+        for expectation in expectations:
+            expectation.register_save_callback(save_callback=self._save_expectation)
+        return expectations
 
     @public_api
     def add(self, expectation: Expectation) -> Expectation:
         """Add an Expectation to the collection."""
-        if not any(
-            expectation.configuration == existing_config
+        if expectation.id:
+            raise RuntimeError(
+                "Cannot add Expectation because it already belongs to an ExpectationSuite. "
+                + "Please make a copy of your Expectation with `copy(expectation)` and try again."
```

Comment:
this requires implementing `Expectation.__copy__`, but i think this is a good way to allow users to reuse expectations in multiple suites. ex:

Reply:
thanks - updated in [77be13d](https://github.com/great-expectations/great_expectations/pull/9125/commits/77be13d67a3a0c3cd7987690a0a011584b11da4c)